### ISSUE TITLE
Does clownface understand the abbriviation `a` instead of `rdf:type`?

### ISSUE THREAD

**Issue by TessavWalstijn** [zazuko/clownface#79]
TLDR: A simple yes or no answer on the title will satisfy me.

Currently having an issue where I get a `text/turtle` response where I get something like this:
```
@prefix schema: <http://schema.org/> .

<https://some.iri.com/for-example-1>
    a            schema:Dataset;
    schema:name  "Nice example name" .

<https://some.iri.com/for-example-2>
    a            schema:Dataset;
    schema:name  "Other example name" .
```

However understanding from examples:
- https://zazuko.github.io/clownface/#/traversal?id=graph-traversal
- https://zazuko.github.io/clownface/#/traversal?id=node-lookup

I see the data used from https://github.com/zazuko/tbbt-ld/
Where the use of `a` in a `text/turle` is used for `rdf:type`:
- https://github.com/zazuko/tbbt-ld/blob/master/data/person/amy-farrah-fowler.ttl 

So I might be not using the following of clownface incorrect in the project
```typescript
data.has(Ns.rdf.Type, Ns.schema.Dataset)
```

Now our simplified code (I do not expect you to look at this)
```typescript
import namespace, { NamespaceBuilder } from '@rdfjs/namespace'
import defaultFormats from '@rdfjs/formats-common'
import fetch from '@rdfjs/fetch'
import rdfExt from 'rdf-ext'
import DatasetExt from 'rdf-ext/lib/Dataset'
import clownface from 'clownface'

class Ns {
  static schema: NamespaceBuilder = namespace('http://schema.org/')
  static rdf: NamespaceBuilder = namespace(
    'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
  )
}

const RdfClientGet = async (
  url: string,
  params?: URLSearchParams,
  contentType = 'text/turtle',
): Promise<DatasetExt> => {
  let newUrl = url
  if (params) newUrl = url + '?' + params
  const response = await fetch(newUrl, {
    method: 'get',
    headers: {
      Accept: contentType,
      Authorization: 'Bearer ' + getToken(),
    },
  })

  if (response?.status == 404) Router.replace('404')

  const format = response.headers.get('content-type') as string
  const parser = defaultFormats.parsers.get(format)
  const dataset = rdfExt.dataset()
  if (response.body) {
    const stream = parser?.import(response.body as any)
    stream?.on('data', (quad) => dataset.add(quad))
    const streamPromise = new Promise((resolve, reject) => {
      stream?.on('end', () => resolve(stream.read))
      stream?.on('error', () => reject)
    })
    await streamPromise
  }

  if (response.status > 299) {
    throw new BadRequest(response, dataset)
  }
  return dataset
}

const getData = () => {
  RdfClientGet('https://example.com/')
    .catch((error) => {
      // custom error stuff
    })
    .then((value) => {
      if (value) {
        const data = clownface({ dataset: value })

        const quads = data.has(Ns.rdf.Type, Ns.schema.Dataset)

        quads.forEach((quad) => {
          const source = {
            iri: quad.term.value,
            name: {
              en: quad.out(Ns.schema('name'), { language: 'en' }).value,
              nl: quad.out(Ns.schema('name'), { language: 'nl' }).value,
            },
          }

          console.log(source)
          // Expected:
          // {
          //   iri: https://some.iri.com/for-example-1 // or -2
          //   name: {
          //     en: 'Nice example name' // or 'Other example name'
          //   }
          // }
          // Result:
          // Never able to loop over quads
          //
        })
      }
    })
}

```

**Comment by tpluscode** [comment]
There is no short answer because the `a` shorthand for `rdf:type` is a feature of turtle serialisation. Clownface only deals with parsed graphs in memory. I suppose NO is the short answer and when accessing the graph with clownface, the `rdf:type` term has to be used.

The problem appears to be the casing. It should be `Ns.rdf.type` (lower case) to match the actual URL 

I looked at you snippet after all, and I might suggest some simplifications. I change `rdf-ext` to `@zazuko/env` because the former has recently removed clownface and we also added some since features, such as common namespaces built-in and type declarations. That also includes type-checking for vocabularies, which would have alerted you about that above. 

Here's how much can can abbreviate but using all library features (most of which is available in the libs you import above, only more tightly packed in `@zazuko/env-node`)

```js
import rdf from '@zazuko/env-node'

const response = await rdf.fetch('http://zazuko.github.io/tbbt-ld/dist/tbbt.nt')
const dataset = await response.dataset()

const data = rdf.clownface({ dataset })
const quads = data.has(rdf.ns.rdf.type, rdf.ns.schema.Person)

quads.forEach((quad) => {
    const source = {
        iri: quad.term.value,
        name: quad.out(rdf.ns.schema.givenName).value
    }

  console.log(source)
})
```

**Comment by TessavWalstijn** [comment]
Thanks! For the look. It was the typo.

TIL about `@zazuko/env` and will look in to it

**Comment by tpluscode** [comment]
tl;dr; for `@zazuko/env` you'd need to add the parsers yourself as they are excluded to reduce dependencies if used in the browser. 

`@zazuko/env-node` includes parsers/serializers and also an additional utility to access the filesystem (https://npm.im/@zazuko/rdf-utils-fs)
